// figma-js/build/module/figmaTypes.js
/**
 * Enum describing how layer blends with layers below
 * This type is a string enum with the following possible values
 */
export var BlendMode;
(function (BlendMode) {
    BlendMode[BlendMode["PASS_THROUGH"] = 0] = "PASS_THROUGH"; /** (Only applicable to objects with children) */
    BlendMode[BlendMode["NORMAL"] = 1] = "NORMAL";
    /** Darken: */
    BlendMode[BlendMode["DARKEN"] = 2] = "DARKEN";
    BlendMode[BlendMode["MULTIPLY"] = 3] = "MULTIPLY";
    BlendMode[BlendMode["LINEAR_BURN"] = 4] = "LINEAR_BURN";
    BlendMode[BlendMode["COLOR_BURN"] = 5] = "COLOR_BURN";
    /** Lighten: */
    BlendMode[BlendMode["LIGHTEN"] = 6] = "LIGHTEN";
    BlendMode[BlendMode["SCREEN"] = 7] = "SCREEN";
    BlendMode[BlendMode["LINEAR_DODGE"] = 8] = "LINEAR_DODGE";
    BlendMode[BlendMode["COLOR_DODGE"] = 9] = "COLOR_DODGE";
    /** Contrast: */
    BlendMode[BlendMode["OVERLAY"] = 10] = "OVERLAY";
    BlendMode[BlendMode["SOFT_LIGHT"] = 11] = "SOFT_LIGHT";
    BlendMode[BlendMode["HARD_LIGHT"] = 12] = "HARD_LIGHT";
    /** Inversion: */
    BlendMode[BlendMode["DIFFERENCE"] = 13] = "DIFFERENCE";
    BlendMode[BlendMode["EXCLUSION"] = 14] = "EXCLUSION";
    /** Component: */
    BlendMode[BlendMode["HUE"] = 15] = "HUE";
    BlendMode[BlendMode["SATURATION"] = 16] = "SATURATION";
    BlendMode[BlendMode["COLOR"] = 17] = "COLOR";
    BlendMode[BlendMode["LUMINOSITY"] = 18] = "LUMINOSITY";
})(BlendMode || (BlendMode = {}));
export var EasingType;
(function (EasingType) {
    EasingType[EasingType["EASE_IN"] = 0] = "EASE_IN"; /** Ease in with an animation curve similar to CSS ease-in */
    EasingType[EasingType["EASE_OUT"] = 1] = "EASE_OUT"; /** Ease out with an animation curve similar to CSS ease-out */
    EasingType[EasingType["EASE_IN_AND_OUT"] = 2] = "EASE_IN_AND_OUT"; /** Ease in and then out with an animation curve similar to CSS ease-in-out */
})(EasingType || (EasingType = {}));
