// figma-js/build/module/index.js
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
export * from './figmaTypes';
import axios from 'axios';
export var Client = function (opts) {
    var headers = opts.accessToken
        ? {
            Authorization: "Bearer " + opts.accessToken
        }
        : {
            'X-Figma-Token': opts.personalAccessToken
        };
    var client = axios.create({
        baseURL: "https://" + (opts.apiRoot || 'api.figma.com') + "/v1/",
        headers: headers
    });
    return {
        client: client,
        file: function (fileId, params) {
            if (params === void 0) { params = {}; }
            return client.get("files/" + fileId, { params: params });
        },
        fileImages: function (fileId, params) {
            return client.get("images/" + fileId, {
                params: __assign({}, params, { ids: params.ids.join(',') })
            });
        },
        comments: function (fileId) { return client.get("files/" + fileId + "/comments"); },
        postComment: function (fileId, params) {
            return client.post("files/" + fileId + "/comments", params);
        },
        teamProjects: function (teamId) { return client.get("teams/" + teamId + "/projects"); },
        projectFiles: function (projectId) { return client.get("projects/" + projectId + "/files"); }
    };
};
